bignum.h

```diff
@@ -25,8 +37,32 @@ typedef struct bignum
 
 /* Init/Printing: */
 void bignum_init(bignum *n);
-void bignum_from_int(bignum *n, unsigned int i);
+void bignum_from_int(bignum *n, BN_DTYPE_TMP i);
 unsigned int bignum_to_int(bignum *n);
 void bignum_from_string(bignum *n, char *str);
 void bignum_to_string(bignum *n, char *str);
-void print_bignum(bignum *n);
\ No newline at end of file
+void print_bignum(bignum *n);
+void bignum_assign(bignum *dst, bignum *src); /* Copy src into dst -- dst := src */
+
+/* Arithmetic operations: */
+void bignum_add(bignum *a, bignum *b, bignum *c);				/* c = a + b */
+void bignum_sub(bignum *a, bignum *b, bignum *c);				/* c = a - b */
+void bignum_mul(bignum *a, bignum *b, bignum *c);				/* c = a * b */
+void bignum_div(bignum *a, bignum *b, bignum *c);				/* c = a / b */
+void bignum_mod(bignum *a, bignum *b, bignum *c);				/* c = a % b */
+void bignum_divmod(bignum *a, bignum *b, bignum *c, bignum *d); /* c = a/b, d = a%b */
+
+/* Bitwise operators: */
```

Comment:
Missing definitions for lshift and the one bit shifts?

Reply:
We didn't need it yet so I haven't included